### ISSUE TITLE
refactor: dynamically retrieve criteria from the new React Context

### ISSUE THREAD

**Issue by GresilleSiffle** [tournesol-app/tournesol#599]
**blocked by** https://github.com/tournesol-app/tournesol/issues/600

---

In order to make the front end compatible with multiple polls, the criteria must be dynamically retrieved from the back end. This task is handled by in #600 and now the components should use this context to display the criteria.

Note some criteria are required and some are optional. Mandatory criteria must not be deselectable in the comparison UI. 

- [ ] In the comparison page, could be in its own PR
- [ ] In the recommendation filters, could be in its own PR

**technical specifications**

How about using React Context, to store the currently selected poll and its citeria.
https://en.reactjs.org/docs/context.html
